pkg/retriever/bitswaphelpers/indexerrouting.go

```diff
@@ -0,0 +1,82 @@
+package bitswaphelpers
+
+import (
+	"context"
+	"sync"
+
+	"github.com/filecoin-project/lassie/pkg/types"
+	"github.com/ipfs/go-cid"
+	routinghelpers "github.com/libp2p/go-libp2p-routing-helpers"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/routing"
+)
+
+var _ routing.Routing = (*IndexerRouting)(nil)
+
+// IndexerRouting provides an interface that satisfies routing.Routing but only returns
+// provider records based on a preset set of providers read from the context key.
+// Bitswap will potentially make multiple FindProvidersAsync requests, and the cid passed will not always be the root
+// As a result, we have to rely on the retrieval id within a context key
+// Also wile there is a delegated routing client that talks to the indexer, we use this cause we run it on
```

Comment:
```suggestion
// Also while there is a delegated routing client that talks to the indexer, we use this cause we run it on
```

sorry for the tiny nit, this is a great comment block otherwise